components/raftstore/src/store/worker/pd.rs

```diff
@@ -1114,6 +1114,9 @@ where
             .region_keys_read
             .observe(region_stat.read_keys as f64);
 
+        // TODO: this should only be done if enabled in config.
+        self.coprocessor_host
+            .on_region_heartbeat(&region, &region_stat);
```

Comment:
follow here to see what happens when a new heartbeat arrives.

Reply:
I think all regions will be considered as `active_regions` since they periodically report their heartbeat.

BTW, we already record the region_stat of active regions in `region_peers`, we can collect the active topN region like https://github.com/tikv/tikv/blob/66847e9c5ac701e73da1366938a7e2306cacc671/components/raftstore/src/store/worker/pd.rs#L1143-L1169.  then we don't need to observe the heartbeat event.